libs/ktem/ktem/reasoning/simple.py

```diff
@@ -711,7 +831,22 @@ def get_pipeline(cls, settings, states, retrievers):
             retrievers: the retrievers to use
         """
         prefix = f"reasoning.options.{cls.get_info()['id']}"
-        pipeline = cls(retrievers=retrievers)
+        example_path = (
+            Path(getattr(flowsettings, "KH_APP_DATA_DIR", ""))
+            / "rephrase_question_train.json"
+        )
+        examples = json.load(open(example_path, "r"))
+        examples = [
+            {"question": item["input"], "rewritten_question": item["output"]}
+            for item in examples
+        ]
```

Comment:
This logic should be handled by your `FewshotRewriteQuestionPipeline`. You are coupling the logic of preparing input for one class to another class which is a bad practice.

Moreover, if you want your `FewshotRewriteQuestionPipeline` class to load examples from a file, then you should define a format for the input file, and include that definition inside the corresponding method's docstring. Loading and then transforming data like this creates a hidden intermediate format and unnecessary complexity.

If this examples list is necessary for parameterising the class, it should be stored along side the class definition as a Python object (like what we do with prompt templates). If the content is too long, storing it as a file in the same module is okay.